### PR TITLE
mgr/orch: Add some more type annotations

### PR DESCRIPTION
**Pull request by sebastian-philipp**
Made `orch.Completion` a generic type

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/cephadm/module.py [hunk 1]
```diff
@@ -1523,19 +1523,19 @@ def remove_daemons(self, names):
...
         self._trigger_preview_refresh(service_name=service_name)
         found = self.spec_store.rm(service_name)
         if found:
             self._kick_serve_loop()
-            return ['Removed service %s' % service_name]
+            return 'Removed service %s' % service_name
         else:
             # must be idempotent: still a success.
-            return [f'Failed to remove service. <{service_name}> was not found.']
+            return f'Failed to remove service. <{service_name}> was not found.'
```
Comment (mgfritch) [inline src/pybind/mgr/cephadm/module.py:1535]:
```suggestion
            return f'Failed to remove service. {service_name} was not found.'
```
minor nit..output looks weird with `<>` chars? remove these?

File: src/pybind/mgr/orchestrator/_interface.py [hunk 2]
```diff
@@ -789,7 +790,7 @@ def resume(self):
         raise NotImplementedError()
 
     def add_host(self, host_spec):
-        # type: (HostSpec) -> Completion
+        # type: (HostSpec) -> Completion[str]
```
Comment (mgfritch) [inline src/pybind/mgr/orchestrator/_interface.py:793]: convert these to py3 style while we are here?

**Comment by sebastian-philipp** [comment]
http://pulpito.ceph.com/swagner-2020-07-28_08:28:33-rados:cephadm-wip-swagner3-testing-2020-07-27-1524-distro-basic-smithi/

**Comment by sebastian-philipp** [comment]
This PR conflicted with `apply_drivegroups()` and thus the rebase only affects `make check`. IMO should be ok to merge.

**Comment by sebastian-philipp** [comment]
jenkins test make check

**Pull request merged**